src/LSST/control/PowerSubsystem.cpp

```diff
@@ -170,6 +285,496 @@ double PowerSubsystemConfig::outputOffMaxDelay() const {
+    // BasePowerOutput->output_should_be_on.vi
+    bool outputIsOn = _powerShouldBeOn();
+
+    double voltage = getVoltage();
+    if (voltage > _psCfg.getMaxVoltageFault()) {
+        LERROR(getClassName(), " voltage(", voltage, ") is too high, turning off");
+        _sendFaultMgrError();
+        _setPowerOff(string(__func__) + "voltage too high");
+        return;
+    }
+    if (voltage > _psCfg.getMaxVoltageWarn()) {
+        LWARN(getClassName(), " voltage(", voltage, ") above warning level ", _psCfg.getMaxVoltageWarn());
+        _sendFaultMgrWarn();
+    }
+
+    util::TIMEPOINT now = util::CLOCK::now();
+    switch (_actualPowerState) {
+    case ON:
+    {
+        if (!outputIsOn) {
+            LDEBUG(_getPowerShouldBeOnStr(), " ON");
+            _setPowerOff(string(__func__) + " output is not on when it should be on");
+            return;
+        }
+
+        if (_checkForPowerOnBreakerFault(voltage)) {
+            LERROR("Breaker fault while _actualPowerState == ON");
+            _setPowerOff(string(__func__) + " Breaker fault _actualPowerState == ON");
+            return;
+        }
+
+        if (_phase <= 1) {
+            LDEBUG(getClassName(), " ON phase 1");
+            // ouput_voltage_is_stable.vi
+            double timeSincePhaseStartInSec = util::timePassedSec(_phaseStartTime, now);
+            double minTimeToStablize = _psCfg.getVoltageSettlingTime() - _psCfg.getBreakerOperatingVoltageRiseTime();
+            if(timeSincePhaseStartInSec > minTimeToStablize) {
+                _phase = 2;
+                _phaseStartTime = util::CLOCK::now();
+                LINFO(getClassName(), " ON phase 2 reached");
+            }
+        } else if (_phase == 2) {
+            if (voltage < _psCfg.getMinVoltageWarn()) {
+                LWARN(getClassName(), " voltage(", voltage, ") below warning level ", _psCfg.getMinVoltageWarn());
+                _sendFaultMgrWarn();
+            }
+            if (voltage < _psCfg.getMinVoltageFault()) {
+                LWARN(getClassName(), " voltage(", voltage, ") below fault level ", _psCfg.getMinVoltageFault());
+                _sendFaultMgrError();
+                _setPowerOff(string(__func__) + " voltage too low");
+            }
+        }
+        break;
+    }
+    case OFF: [[fallthrough]];
+    case TURNING_OFF:
+        _actualPowerState = TURNING_ON;
+        _phase = 1;
+        _phaseStartTime = now;
+        [[fallthrough]];
+    case TURNING_ON:
+    {
+        double timeSincePhaseStartInSec = util::timePassedSec(_phaseStartTime, now);
+        LDEBUG(getClassName(), " TURNING_ON phase=", _phase, " timeInPhase=", timeSincePhaseStartInSec,
+               " telemCount=", _telemCounter);
+        if (_phase == 1) {
+            // Presumably, waiting for the signal to stabilize.
+            ++_telemCounter;
+            // Value of 10 comes from telemetry_is_stable.vi, no explanation.
+            if (_telemCounter >= 10) {
```

Comment:
10 DAQ reads since this `_phase` started. There was no explanation why then needed to wait for 10 DAQ reads, but I doubt they would have put it there if it worked without it.